### PR TITLE
Endpoint /api/books/{book_id} includes book reviews in response

### PR DESCRIPTION
**Pull request by NenadBecanovicPD**

### CODE REVIEW

File: src/test/java/com/productdock/library/gateway/service/BookApiTest.java [hunk 1]
```diff
@@ -15,15 +15,17 @@
 
 import java.io.IOException;
 
+//import static com.productdock.library.gateway.data.provider.ReviewDtoMother.defaultReviewDto;
```
Comment (vukomanv-pd) [inline src/test/java/com/productdock/library/gateway/service/BookApiTest.java:18]: Remove unused import

File: src/test/java/com/productdock/library/gateway/service/BookDetailsResponseCombinerShould.java [hunk 2]
```diff
@@ -21,28 +18,35 @@ class BookDetailsResponseCombinerShould {
...
+        anyRecord.put("recordProperty", "recordValue");
+        List<Object> rentalRecordsDto = List.of(anyRecord);
         var availableBookCount = 0;
 
-        var bookDetails = bookDetailsResponseCombiner.generateBookDetailsDto(bookDto,rentalRecordsDto,availableBookCount);
+        var bookDetails = bookDetailsResponseCombiner.generateBookDetailsDto(anyDto, rentalRecordsDto, availableBookCount);
 
-        assertThat(bookDetails.records).isEqualTo(rentalRecordsDto);
+        assertThat(bookDetails.get("records")).isNotNull();
+        assertThat(bookDetails.get("records").size()).isEqualTo(1);
```
Comment (milicazivkov85) [inline src/test/java/com/productdock/library/gateway/service/BookDetailsResponseCombinerShould.java:50]: Try hasSize

File: src/test/java/com/productdock/library/gateway/service/BookServiceShould.java [hunk 3]
```diff
@@ -8,28 +9,28 @@
...
 
 @ExtendWith(MockitoExtension.class)
 class BookServiceShould {
 
     private static final String BOOK_ID = "1";
     private static final String JWT_TOKEN = "";
     private static final int AVAILABLE_BOOK_COUNT = 1;
-    private static final Mono<BookDto> BOOK_DTO_MONO = Mono.just(defaultBookDto());
-    private static final Mono<List<BookRentalRecordDto>> RENTAL_RECORDS_DTO_MONO = Mono.just(Arrays.asList(defaultBookRentalRecordDto()));
+    public static final Object CATALOG_RESPONSE = Mockito.mock(Object.class);
```
Comment (milicazivkov85) [inline src/test/java/com/productdock/library/gateway/service/BookServiceShould.java:28]: private

File: src/test/java/com/productdock/library/gateway/service/BookDetailsResponseCombinerShould.java (hunk 2 above)
Comment (NenadBecanovicPD) [reply src/test/java/com/productdock/library/gateway/service/BookDetailsResponseCombinerShould.java:50]: I tried.

**Pull request merged**